src/types.ts

```diff
@@ -281,9 +297,15 @@ export interface IDataInput {
   data?: string;
 }
 
+interface IDataTokenCreationTx {
+  name: string;
+  symbol: string;
+  tokenVersion: TokenVersion;
+}
+
 // XXX: This type is meant to be used as an intermediary for building transactions
 // It should have everything we need to build and push transactions.
-export interface IDataTx {
+export interface IDataTx extends Partial<IDataTokenCreationTx> {
```

Comment:
Why is it partial? Don't you need all 3 fields?

I think I prefer duplicating the fields in this case, it's not explicit which fields are optional, or which fields are included, since it's partial